src/components/Card.js

```diff
@@ -8,27 +8,60 @@
 // import PopupWithImage from "../scripts/PopupWithImage.js";
 
 export default class Card {
-  constructor({ name, link }, cardSelector, handleImageClick) {
-    this._name = name;
-    this._link = link;
+  constructor(
+    cards,
+    cardSelector,
+    handleImageClick,
+    handleCardDeleteClick,
+    handleCardLike,
+    _handleCardLikeRemove
+  ) {
+    this._name = cards.name;
+    this._link = cards.link;
+    this._id = cards._id;
+    this._isLiked = cards.isLiked;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleCardDeleteClick = handleCardDeleteClick;
+    this._handleCardLike = handleCardLike;
+    this._handleCardLikeRemove = _handleCardLikeRemove;
+  }
+  getId() {
+    return this._id;
   }
   _setEventListeners() {
     // ---- card like button -----------------------------------------------------------------
 
     this._cardElement
       .querySelector(".card-heart")
       .addEventListener("click", () => {
-        this._handleLikeIcon();
+        if (this._isLiked) {
+          this._handleCardLikeRemove(this);
+          this._isLiked = !this._isLiked;
```

Comment:
`          this._isLiked = !this._isLiked;
`

You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.